test/visreg/visreg.ts

```diff
@@ -46,6 +47,22 @@ export function suite(browserName: string) {
         await snappit.snap("{browserName}/nav/custom-elements", $(util.selectors.nav));
     });
 
+    // Test Suite for Modal for Firefox browser
+    test("components/modals", async t => {
+        await util.go(driver, "components/modals");
+        await snappit.snap("{browserName}/modals/modals", $(util.selectors.modal));
+    });
+
+    test("components/open-modals", async t => {
+        await util.go(driver, "components/modals");
+        await modalTest(t, snappit, driver, "Open Modal", util.selectors.openModal);
+    });
+
+    test("components/close-modals", async t => {
+        await util.go(driver, "components/modals");
+        await modalTest(t, snappit, driver, "Close Modal", util.selectors.closeModal);
```

Comment:
If you drop the `modalTest` abstraction, these test definitions become much simpler.

One test that should click the open modal button. Take a picture. Assert that the modal is open.

Another test the should click the open modal button. Then, it clicks the close modal button. Take another picture. Assert that the modal is closed.

Reply:
Thank you Andrew. Followed the above idea recommended by you.